### PR TITLE
2022.3 oneAPI release promotion

### PR DESCRIPTION
**Pull request by akhursev**
@rscohn2, please, review

### CODE REVIEW

File: var/spack/repos/builtin/packages/intel-oneapi-dal/package.py [hunk 1]
```diff
@@ -29,6 +29,12 @@ class IntelOneapiDal(IntelOneApiLibraryPackage):
     )
 
     if platform.system() == "Linux":
+        version(
+            "2021.7",
```
Comment (rscohn2) [inline var/spack/repos/builtin/packages/intel-oneapi-dal/package.py]: package number is 2021.7.0 in the download URL?

File: var/spack/repos/builtin/packages/intel-oneapi-ipp/package.py [hunk 2]
```diff
@@ -30,6 +30,12 @@ class IntelOneapiIpp(IntelOneApiLibraryPackage):
     )
 
     if platform.system() == "Linux":
+        version(
+            "2021.7",
```
Comment (rscohn2) [inline var/spack/repos/builtin/packages/intel-oneapi-ipp/package.py]: package number is 2021.6.1 in download url?

**Review by rscohn2 — CHANGES_REQUESTED** [review]
Looks like version numbers are not correct. Offline discussion about the other packages.

**Comment by rscohn2** [comment]
Tests are failing: https://github.com/rscohn2/oneapi-spack-tests/pull/35. It may be issues in the tests. I will wait until all the packages are there and version numbers are confirmed.

File: var/spack/repos/builtin/packages/intel-oneapi-dal/package.py (hunk 1 above)
Comment (akhursev) [reply var/spack/repos/builtin/packages/intel-oneapi-dal/package.py]: Yes, it was RTMed

File: var/spack/repos/builtin/packages/intel-oneapi-ipp/package.py (hunk 2 above)
Comment (akhursev) [reply var/spack/repos/builtin/packages/intel-oneapi-ipp/package.py]: The package number is correct. I do believe we have a problem with the version.

**Review by rscohn2 — APPROVED** [review]

**Comment by rscohn2** [comment]
@akhursev: I pushed some changes to version number and corrected the issues that ``spack style`` was flagging. Please look this over before I merge.

**Comment by rscohn2** [comment]
Tests passed: https://github.com/rscohn2/oneapi-spack-tests/actions/runs/3127317028/jobs/5073809204

**Pull request merged**